### DISCUSSION TITLE
Passing variables to blocks

### DISCUSSION THREAD

**Discussion by marlonbasten** [area17/twill#982] [Q&A] [answered]
Is there a way to pass variables to blocks?
At the moment I have to use the @php @endphp tags in the views directly.
If I call the renderBlocks() method in a view and the view has the variables the blocks need, I still get a variable not found error.

**Comment by pboivin** [comment, marked as answer]
Hey @marlonbasten,

I think you may be looking for the [$data argument of renderBlocks().](https://twill.io/docs/api/2.x/A17/Twill/Models/Behaviors/HasBlocks.html#method_renderBlocks)

You can use it to pass extra data to your blocks, like so :

```
@php
  $context = $page->getContext();
@endphp

{!! $page->renderBlocks(true, [], ['context' => $context]) !!}
```

Also, you may be interested in [Laravel's View::share()](https://laravel.com/docs/8.x/views#sharing-data-with-all-views), that would allow you to make certain variables globally accessible to all your Blade views (including custom blocks).

Hope this can help!
